internal/pkg/mining/scheduler.go

```diff
@@ -7,108 +7,127 @@ package mining
 
 import (
 	"context"
+	"fmt"
 	"sync"
 	"time"
 
 	"github.com/filecoin-project/go-filecoin/internal/pkg/block"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/clock"
+	th "github.com/filecoin-project/go-filecoin/internal/pkg/testhelpers"
 	"github.com/pkg/errors"
 )
 
-// Scheduler is the mining interface consumers use. When you Start() the
-// scheduler it returns a channel and a sync.WaitGroup:
-//   - channel: the scheduler emits outputs (mined blocks) to the caller on this channel
-//   - waitgroup: signals that the scheduler and any goroutines it launched
-//             have stopped. (Context cancelation happens async, so you
-//             need some way to know when it has actually stopped.)
-//
-// Once started, a scheduler can be stopped by canceling the context provided to `Start()`,
-// which will signal on the waitgroup when it's actually done.
+// Scheduler is the mining interface consumers use.
 type Scheduler interface {
-	Start(miningCtx context.Context) (<-chan Output, *sync.WaitGroup)
+	Start(miningCtx context.Context, skipCh chan bool) (<-chan Output, *sync.WaitGroup)
```

Comment:
Why is this a channel on the start and not a `Pause()`/`Resume()` on the interface itself?

Reply:
That sounds like a better idea.  I might switch over.

Reply:
Currently deferring this work to the PR that will hook up syncer state changes to mining scheduler pausing